python-checks/src/main/java/org/sonar/python/checks/UnreachableExceptCheck.java

```diff
@@ -77,23 +79,39 @@ private static void handleExceptClause(SubscriptionContext ctx, Map<String, Expr
 
   private static void handleExceptionExpression(SubscriptionContext ctx, Map<String, Expression> caughtTypes,
                                          Expression exceptionExpression, Map<String, Expression> caughtInExceptClause) {
-    if (exceptionExpression instanceof HasSymbol) {
-      Symbol symbol = ((HasSymbol) exceptionExpression).symbol();
-      if (symbol != null && symbol.kind().equals(Symbol.Kind.CLASS)) {
-        ClassSymbol classSymbol = (ClassSymbol) symbol;
-        List<Expression> handledExceptions = retrieveAlreadyHandledExceptions(classSymbol, caughtTypes);
+    Optional.of(exceptionExpression)
```

Comment:
I don't really mind using a more functional style, especially for the newly added methods. On the other hand, I feel like changes are easier to write and review when we keep the style consistent (when it makes sense). In this case, I feel like an implementation like the following would have maintained the idiomatic style of the existing method:
```
private static void handleExceptionExpression(SubscriptionContext ctx, Map<String, Expression> caughtTypes,
                                         Expression exceptionExpression, Map<String, Expression> caughtInExceptClause) {
    if (!(exceptionExpression instanceof HasSymbol)) {
      return;
    }
    Symbol symbol = ((HasSymbol) exceptionExpression).symbol();
    if (symbol == null) {
      return;
    }
    List<Expression> handledExceptions = symbol.kind().equals(Symbol.Kind.CLASS) ? retrieveAlreadyHandledExceptionsByClass((ClassSymbol) symbol, caughtTypes)
      : retrieveAlreadyHandledExceptionsByFullyQualifiedName(symbol, caughtTypes);
    if (!handledExceptions.isEmpty()) {
      PreciseIssue issue = ctx.addIssue(exceptionExpression, "Catch this exception only once; it is already handled by a previous except clause.");
      handledExceptions.forEach(h -> issue.secondary(h, SECONDARY_MESSAGE));
    }
    caughtInExceptClause.put(symbol.fullyQualifiedName(), exceptionExpression);
  }
```

This approach would also have the benefit of avoiding the IDE warning about `symbol` being nullable on line 88 (despite the check on the `Objects.nonNull(expression.symbol())`, we still have the warning as we call the API again with the `CheckForNull` annotation) and also avoiding the somewhat redundant cast to `Expression` on line 94 (Since we started with an `Expression`, it should still be that despite the cast in the mapping operation - but the compiler doesn't know that).

I let you decide if you want to update the implementation. If you do, don't forget to remove the `java.util.Objects` import which would be unused as a result.

Reply:
done